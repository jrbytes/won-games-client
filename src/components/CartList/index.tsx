import Button from 'components/Button'
import GameItem, { GameItemProps } from '../GameItem'

import * as S from './styles'

export type CartListProps = {
  items: GameItemProps[]
  total: string
  hasButton?: boolean
}

const CartList = ({ items, total, hasButton = false }: CartListProps) => (
  <S.Wrapper>
    {items.map((item) => (
      <GameItem key={item.title} {...item} />
    ))}

    <S.Footer>
      {!hasButton && <span>Total:</span>}
      <S.Total>{total}</S.Total>
      {hasButton && (
        <Button href="/cart" as="a">
          Buy it now
        </Button>
      )}
    </S.Footer>
  </S.Wrapper>
)

export default CartList
