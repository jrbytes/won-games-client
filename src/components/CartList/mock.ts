const mock = [
  {
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    title: 'Red Dead Redemption 2',
    price: 'R$ 235,00'
  },
  {
    img: 'https://source.unsplash.com/user/willianjusten/151x70',
    title: 'Bordelands 3',
    price: 'R$ 235,00'
  }
]

export default mock
