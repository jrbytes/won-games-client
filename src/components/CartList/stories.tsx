import { ComponentStory, ComponentMeta } from '@storybook/react'
import CartList from '.'

import mockItems from './mock'

export default {
  title: 'CartList',
  component: CartList,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  args: {
    items: mockItems,
    total: 'R$ 430,00'
  }
} as ComponentMeta<typeof CartList>

export const Default: ComponentStory<typeof CartList> = (args) => (
  <div style={{ maxWidth: 800, margin: '0 auto' }}>
    <CartList {...args} />
  </div>
)

export const WithButton: ComponentStory<typeof CartList> = (args) => (
  <div style={{ maxWidth: 800, margin: '0 auto' }}>
    <CartList {...args} hasButton />
  </div>
)

export const Empty: ComponentStory<typeof CartList> = () => (
  <div style={{ maxWidth: 800, margin: '0 auto' }}>
    <CartList />
  </div>
)
