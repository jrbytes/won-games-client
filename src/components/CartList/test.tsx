import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CartList from '.'
import mockItems from './mock'

describe('<CartList />', () => {
  it('should be able to render the cart list', () => {
    const { container } = renderWithTheme(
      <CartList items={mockItems} total="R$ 470" />
    )

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText(/R\$ 470/i)).toHaveStyle({ color: '#F231A5' })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render the button', () => {
    renderWithTheme(<CartList items={mockItems} total="R$ 470" hasButton />)

    expect(screen.getByText(/buy it now/i)).toBeInTheDocument()
  })
})
