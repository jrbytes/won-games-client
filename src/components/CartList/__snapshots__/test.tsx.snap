// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should be able to render the cart list 1`] = `
.c1 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c3 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c6 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c7 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Red Dead Redemption 2"
          data-nimg="1"
          decoding="async"
          height="70"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Red Dead Redemption 2
        </h3>
        <div
          class="c6"
        >
          $ 
          R$ 235,00
        </div>
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Bordelands 3"
          data-nimg="1"
          decoding="async"
          height="70"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Fuser%2Fwillianjusten%2F151x70&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Bordelands 3
        </h3>
        <div
          class="c6"
        >
          $ 
          R$ 235,00
        </div>
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    <span>
      Total:
    </span>
    <span
      class="c8"
    >
      R$ 470
    </span>
  </div>
</main>
`;
